### PR TITLE
379 preview repeat

### PR DESCRIPTION
**Pull request by phyzical**
* implemented previewRepeat to repeat preview animation
* implemented previewRepeatDelay to change the delay between repeats
* fixed a type
* removed some unused imports
* updated ts
* updated readmes for new logic
* refactor animation into its own function
* added logic to remove preview repeat on change of previewRepeat

### CODE REVIEW

File: SwipeListExample/example.js [hunk 1]
```diff
@@ -10,7 +10,8 @@ import {
     View,
 } from 'react-native';
 
-import { SwipeListView, SwipeRow } from 'react-native-swipe-list-view';
```
Comment (jemise111) [inline SwipeListExample/example.js]: please revert this

File: components/SwipeRow.js [hunk 2]
```diff
@@ -161,17 +170,35 @@ class SwipeRow extends Component {
...
-            ).start(() => {
-                this.getPreviewAnimation(0, PREVIEW_CLOSE_DELAY).start();
-            });
+            if (this.props.previewRepeat) {
+                const totalRepeatDelay =
+                    (this.props.previewDuration +
+                        this.props.previewOpenDelay +
+                        PREVIEW_CLOSE_DELAY) *
+                        2.5 +
+                    this.props.previewRepeatDelay;
```
Comment (jemise111) [inline components/SwipeRow.js]: Can you please explain this calculation? In testing this I'm noticing the first preview takes a very long time to start. Using the defaults the initial delay would be (300 + 700 + 300) * 2.5 =  3250 ms or 3.25 seconds which seems too long to me. What do you think?

File: types/index.d.ts [hunk 3]
```diff
@@ -278,10 +286,19 @@ interface IPropsSwipeListView<T> {
...
+	previewRepeat: boolean,
+	/**
+	 * Time between each full completed animation in milliseconds
+	 * Default: 1000 (1 second)
+	 */
+	previewRepeatDelay: number,
 	/**
 	 * Delay of the slide out preview animation (milliseconds) // default 700ms
 	 */
-	prewiewOpenDelay: number;
```
Comment (jemise111) [inline types/index.d.ts]: 👀 nice catch

File: SwipeListExample/example.js (hunk 1 above)
Comment (phyzical) [reply SwipeListExample/example.js]: opps must have been my idea, reverted

File: components/SwipeRow.js (hunk 2 above)
Comment (phyzical) [reply components/SwipeRow.js]: yeah there was a few things wrong with this, ill explain in response with commit

File: types/index.d.ts (hunk 3 above)
Comment (phyzical) [reply types/index.d.ts]: ;)

**Comment by phyzical** [comment]
summary:

* reverted changes
* refactored total time delay into a state on prop change to clean it up a bit
* `prevState.timeBetweenPreviewRepeats =
            nextProps.previewDuration * 2 +
            nextProps.previewOpenDelay +
            PREVIEW_CLOSE_DELAY +
            nextProps.previewRepeatDelay;` So we take the duration that applied on open then close, so * 2 and then we need to add the open and close delay to get the "total animation timing" then we add the prop for the repeat delay on the end.
* moved out an initial call to animation as i derped and thought settimeout fires THEN timer kicks in. it does not opps.

I agree the first implementation wasnt right, i missed the previewDuration being applied on open then close rather than being so magic for the total open + close. So now  300 + (700 * 2) + 300 = 2000 ms + delay prop. As far as i can tell it works well now.

Let me know if you notice any other issues

Thanks!

**Pull request merged**